go.mod

```diff
@@ -25,18 +25,18 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/vishvananda/netlink v1.1.1-0.20211101163509-b10eb8fe5cf6
 	github.com/vishvananda/netns v0.0.0-20200728191858-db3c7e526aae
-	go.uber.org/zap v1.25.0
-	golang.org/x/time v0.3.0
+	go.uber.org/zap v1.26.0
+	golang.org/x/time v0.5.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.28.3
-	k8s.io/apiextensions-apiserver v0.28.3
-	k8s.io/apimachinery v0.28.3
-	k8s.io/client-go v0.28.3
-	k8s.io/code-generator v0.28.3
-	k8s.io/kubectl v0.28.3
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
-	sigs.k8s.io/controller-runtime v0.16.3
+	k8s.io/api v0.29.0
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.0
+	k8s.io/client-go v0.29.0
+	k8s.io/code-generator v0.29.0
+	k8s.io/kubectl v0.29.0
+	k8s.io/utils v0.0.0-20231127182322-b307cd553661
+	sigs.k8s.io/controller-runtime v0.16.1-0.20231215020716-1b80b9629af8
```

Comment:
i dont like this. controller runtime is a pretty central pkg we should use release versions.

can this wait until proper release of controller runtime is made ?

Reply:
I understand your concern but there is a deadline to get this in pretty soon. We don't control the release cadence of the controller-runtime. I'm sure they will tag a commit with a release version at some point in the future, but there is no way for us to control that easily.

I prefer to bump the controller runtime, and we can bump again when there is a release. If you want, I will take a note to watch for a release and then I'll bump it again when there is an official tag.

Reply:
alternative is to replace otel pkg, maybe less risky ?

Reply:
Can you elaborate? Do you mean a replace section in go.mod?

Reply:
yep that can be a good idea but before we go that wait I was looking in the code did you find a place where we use the opentelemetry?

even on the vendor folder I was not able to find a place where we call the function that is effected by the CVE

Reply:
`go mod graph | grep otel@`

You can see that the library is pulled in indirectly. It's not that we use otel, rather one of our dependencies uses it

Reply:
right but I searched in the code and was not able to find anything...
can you try to find it and see if we even use it.

maybe it's about metrics and we disable that I the operator

Reply:
Since we don't have a stable version for controller-runtime, we have decided to not bump controller-runtime just yet. I will make this PR instead focus on bumping golang only.